schematron/Consolidation.sch

```diff
@@ -501,23 +501,23 @@ Schematron generated from Trifolia on 7/20/2021
-      <sch:assert id="a-81-7157" test="count(cda:given) &gt; 0">SHALL contain at least one [1..*] given (CONF:81-7157).</sch:assert>
-      <sch:assert id="a-81-7159" test="count(cda:family)=1">SHALL contain exactly one [1..1] family (CONF:81-7159).</sch:assert>
-      <sch:assert id="a-81-7278-c" test="not(tested)">**SHALL NOT** have mixed content except for white space (CONF:81-7278).</sch:assert>
+      <sch:assert id="a-81-7157-c" test="count(cda:given) &gt; 0 or @nullFlavor">SHALL contain at least one [1..*] given (CONF:81-7157).</sch:assert>
+      <sch:assert id="a-81-7159-c" test="count(cda:family)=1 or @nullFlavor">SHALL contain exactly one [1..1] family (CONF:81-7159).</sch:assert>
+      <sch:assert id="a-81-7278-c" test="not(text()[normalize-space()])">**SHALL NOT** have mixed content except for white space (CONF:81-7278).</sch:assert>
     </sch:rule>
-    <sch:rule id="r-urn-oid-2.16.840.1.113883.10.20.22.5.1-errors" context="cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.14']]/cda:participant/cda:associatedEntity/cda:associatedPerson/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.14' and @extension='2015-08-01']]/cda:participant/cda:associatedEntity/cda:associatedPerson/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.1' and @extension='2015-08-01']]/cda:recordTarget/cda:patientRole/cda:patient/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.1' and @extension='2015-11-02']]/cda:recordTarget/cda:patientRole/cda:patient/cda:name | cda:author[cda:templateId[@root='2.16.840.1.113883.10.20.37.4.2' and @extension='2017-08-01']]/cda:assignedAuthor/cda:assignedPerson/cda:name">
+    <sch:rule id="r-urn-oid-2.16.840.1.113883.10.20.22.5.1-errors" context="cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.14']]/cda:participant/cda:associatedEntity/cda:associatedPerson/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.14' and @extension='2015-08-01']]/cda:participant/cda:associatedEntity/cda:associatedPerson/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.1' and @extension='2015-08-01']]/cda:recordTarget/cda:patientRole/cda:patient/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.1' and @extension='2015-11-02']]/cda:recordTarget/cda:patientRole/cda:patient/cda:name | cda:author[cda:templateId[@root='2.16.840.1.113883.10.20.37.4.2' and @extension='2017-08-01']]/cda:assignedAuthor/cda:assignedPerson/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.5.1.1.3' and @extension='2019-08-01']]/cda:recordTarget/cda:patientRole/cda:patient/cda:name | cda:ClinicalDocument[cda:templateId[@root='2.16.840.1.113883.10.20.22.1.1.999999' and @extension='Example']]/cda:recordTarget/cda:patientRole/cda:patient/cda:name">
```

Comment:
Yeah - totally. That's what I meant - I'm not manually going through these and trying to figure out what's real and what's false. These 999999 templateIds definitely look like test data (especially with that extension), but I can at least see instances of 999999 in Trifolia, so at least some version of some guide _might_ have included them (one would hope they'd be caught before actual publication). Meanwhile, other templateIds that look legit, I can't find at all in Trifolia (but that's not surprising - I likely don't have access to those IG's.

Ultimately, it would be best if Trifolia didn't export any contexts that _don't_ belong to the IG. But barring that, I'd prefer to not sift through this giant context and validate that every templateId belongs to C-CDA.

Reply:
Ahh, okay. Thanks for clarifying!